nucypher/blockchain/eth/interfaces.py

```diff
@@ -442,6 +442,10 @@ def __log_transaction(self, transaction_dict: dict, contract_function: ContractF
         # Format
         if tx.get('to'):
             tx['to'] = to_checksum_address(contract_function.address)
+        try:
+            tx['selector'] = contract_function.selector
+        except AttributeError:
+            pass
```

Comment:
try / except / pass is always an eyebrow-raiser for me - Here's an alternate approach:

- Relocate `BlockchainInterface.__log_transaction` to `EthereumClient` and `Signer` methods .
- Override a base abstract method to handle any local attributes or functionality